src/fmu/dataio/dataio.py

```diff
@@ -474,8 +474,9 @@ def __post_init__(self) -> None:
                 "disable this warning, remove the 'FMU_DATAIO_CONFIG' env.",
             )
 
-        # global config which may be given as env variable -> a file; will override
-        if GLOBAL_ENVNAME in os.environ:
+        # global config which may be given as env variable
+        # will only be used if not explicitly given as input
+        if not self.config and GLOBAL_ENVNAME in os.environ:
```

Comment:
I wonder if a chain map might make a bit more sense here?

https://docs.python.org/3/library/collections.html#collections.ChainMap

```suggestion
self.config = dict(collections.ChainMap(self.config, os.environ.get(GLOBAL_ENVNAME, {}))
```

Reply:
the environment variable is just pointing to a yaml-file, so we need to enter the function to load this yaml file.
I guess that would be possible to have in this chainmap, but it just does not read as easily. I'll keep as is 👍